### PR TITLE
[dev] Comment in block are ignored

### LINKED ISSUE #503: should not process comment scripts?
When I comment some sciprts in index.html:

```
<!-- build:js({.tmp,client}) app/app.js -->
<script src="app/app.js"></script>

<!-- Mock Code
<script src="components/mock.js"></script>
-->

<!-- endbuild -->
```

I use useminPrepare to generate the config, and then run the generated tasks. But the concated file will include the commented scripts `components/mock.js`, this is a bug?

But when I rewite the comment as a line:

```
<!-- Mock Code <script src="components/mock.js"></script>-->
```

The commented file will not included in concated file.

My config is:

```
// Reads HTML for usemin blocks to enable smart builds that automatically
    // concat, minify and revision files. Creates configurations in memory so
    // additional tasks can operate on them
    useminPrepare: {
      html: ['<%= yeoman.app %>/index.html'],
      options: {
        dest: '<%= yeoman.dist %>/public'
      }
    },

    // Performs rewrites based on rev and the useminPrepare configuration
    usemin: {
      html: ['<%= yeoman.dist %>/public/{,*/}*.html'],
      css: ['<%= yeoman.dist %>/public/{,*/}*.css'],
      js: ['<%= yeoman.dist %>/public/{,*/}*.js'],
      options: {
        assetsDirs: [
          '<%= yeoman.dist %>/public',
          '<%= yeoman.dist %>/public/app',
          '<%= yeoman.dist %>/public/assets/images'
        ],
        // This is so we update image references in our ng-templates
        patterns: {
          js: [
            [/(assets\/images\/.*?\.(?:gif|jpeg|jpg|png|webp|svg))/gm, 'Update the JS to reference our revved images']
          ]
        }
      }
    },
```

### PR DESCRIPTION
**Pull request by stephanebachelier**
Fix #503

### CODE REVIEW

File: test/test-file.js [hunk 1]
```diff
@@ -16,8 +16,20 @@ describe('File', function () {
...
+    assert.equal('foo.css', file.blocks[0].dest);
+    assert.equal(2, file.blocks[0].src.length);
+    assert.equal(4, file.blocks[0].raw.length);
+    assert.equal(' ', file.blocks[0].indent);
+    assert.equal(32, file.blocks[0].indent.charCodeAt(0));
+  });
+
   it('should skip blank lines', function () {
-    var filename = __dirname + '/fixtures/block_with_empty_line.html';
+    var filename = path.join(__dirname + '/fixtures/block_with_empty_line.html');
```
Comment (sindresorhus) [inline test/test-file.js]: `var filename = path.join(__dirname, '/fixtures/block_with_empty_line.html');`

File: test/test-file.js [hunk 2]
```diff
@@ -16,8 +16,20 @@ describe('File', function () {
     assert.ok(file.blocks.length, 2);
   });
 
+  it('should ignore comments', function () {
+    var filename = path.join(__dirname + '/fixtures/block_with_comment.html');
```
Comment (sindresorhus) [inline test/test-file.js]: `var filename = path.join(__dirname, '/fixtures/block_with_comment.html');`

File: test/test-file.js [hunk 3]
```diff
@@ -16,8 +16,20 @@ describe('File', function () {
...
+    var file = new File(filename);
+    console.log(file.blocks);
```
Comment (sindresorhus) [inline test/test-file.js]: ?

Comment (stephanebachelier) [inline test/test-file.js]: I'm stupid :)

**Comment by arthurvr** [comment]
:+1:

**Comment by arthurvr** [comment]
@stephanebachelier was this good to go?

**Comment by stephanebachelier** [comment]
@arthurvr need to check.

**Comment by MichielDeMey** [comment]
Huzzah for this :+1:

**Pull request merged**